src/NuGet.Core/NuGet.Protocol/Converters/MetadataStringOrArrayConverter.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+
+namespace NuGet.Protocol
+{
+    public class MetadataStringOrArrayConverter : JsonConverter
```

Comment:
Does this need to be public?

If Ctrl + F didn't betray me, I think it's only used it Protocol. 

Nice fix to the problem btw :)_

Reply:
Only our parsing of `PackageSearchMetadata` depends on it.

All our JSON converters are public, as well as the metadata type which will be utilizing this converter. Can you help me understand the risk of it remaining public?

Reply:
I'm not sure risk is the correct word. 

https://github.com/NuGet/NuGet.Client/blob/dev/docs/nuget-sdk.md#changes-to-apis 

> Do all of your public APIs need to be public, or can any be made internal?

In general, we try to minimize public APIs because as said in the issue, we have to support them forever. 
Personally, if we could take it back, I'd say all converters should have remained internal.